docarray/index/backends/voyager.py

```diff
@@ -0,0 +1,263 @@
+
+        # Explicitly specify the type of distances to List[float]
+        distances_list = distances.tolist()  # Assuming distances is a numpy array
+
+        return _FindResultBatched(documents, distances_list)
+
+    def _find(
+        self, query: np.ndarray, limit: int, search_field: str = ''
+    ) -> _FindResult:
+        query_batched = np.expand_dims(query, axis=0)
+        batched_result = self._find_batched(
+            queries=query_batched, limit=limit, search_field=search_field
+        )
+
+        # Assuming scores are available in batched_result
+        scores = batched_result.scores
+
+        return self._FindResult(documents=batched_result.documents, scores=scores)
+
+    def _filter(
+        self,
+        filter_query: Any,
+        limit: int,
+    ) -> DocList:
+        result = self.execute_query(filter_query)
+
+        ids = [doc['id'] for doc in result]
+        embeddings = [doc['embedding'] for doc in result]
+
+        docs = DocList.__class_getitem__(cast(Type[BaseDoc], self.out_schema))()
+        for id_, embedding in zip(ids, embeddings):
+            doc = self._doc_from_bytes(embedding)  # You need to implement this method
+            doc.id = id_
+            docs.append(doc)
+
+        return docs
+
+    def _filter_batched(
+        self,
+        filter_queries: Any,
+        limit: int,
+    ) -> List[DocList]:
+        # You can implement batched filtering logic here
+        # For example, execute each filter query separately and combine the results
+        raise NotImplementedError(
+            f'{type(self)} does not support filter-only batched queries.'
+            f' To perform post-filtering on a query, use'
+            f' `build_query()` and `execute_query()`.'
+        )
+
+    def _text_search(
+        self,
+        query: str,
+        limit: int,
+        search_field: str = '',
+    ) -> _FindResult:
+        result = self.execute_query({'text_search': query, 'limit': limit})
+
+        ids = [doc['id'] for doc in result]
+        embeddings = [doc['embedding'] for doc in result]
+
+        docs = DocList.__class_getitem__(cast(Type[BaseDoc], self.out_schema))()
+        for id_, embedding in zip(ids, embeddings):
+            doc = self._doc_from_bytes(embedding)  # You need to implement this method
+            doc.id = id_
+            docs.append(doc)
+
+        return _FindResult(
+            documents=docs,
+            scores=[1.0] * len(docs),  # You may adjust the scores as needed
+        )
+
+    def _text_search_batched(
+        self,
+        queries: Sequence[str],
+        limit: int,
+        search_field: str = '',
+    ) -> _FindResultBatched:
+        # You can implement batched text search logic here
+        # For example, execute each text search query separately and combine the results
+        raise NotImplementedError(
+            f'{type(self)} does not support text search batched queries.'
+        )
+
+
+class NdArray:
```

Comment:
why do u need this?

Reply:
The `NdArray` class was created to provide a versatile and efficient solution for managing multidimensional arrays in Python. It aims to enhance efficiency, enabling optimized storage and access compared to standard Python data structures. The class also supports mathematical operations relevant to scientific computing, machine learning, and data analysis. Additionally, it offers custom functionality, including methods for reshaping, slicing, and applying filters. The abstraction of data representation enhances code readability, and seamless integration with libraries like NumPy ensures compatibility. 

If you want I can remove it completely.

Reply:
I am removing this.

Reply:
Please remove this